src/utils.ts

```diff
@@ -113,9 +113,49 @@ function readCompilerOptions(configPath: string) {
   return parsedConfig.options;
 }
 
+function getStartDir(): string {
+  // This is needed because of the way our tests are structured.
+  // If this is being executed as a library (under node_modules)
+  // we want to start with the project directory that's three
+  // levels above.
+  // If t his is being executed from the test suite, we want to start
+  // in the directory of the test
+
+  const grandparent = path.resolve(__dirname, '..', '..');
+  if (grandparent.endsWith('/node_modules')) {
+    return path.resolve(grandparent, '..');
+  }
+  return '.';
+}
+
+function getPathToClosestTSConfig(
+  startDir?: string,
+  previousDir?: string,
+): string {
+  // Starting with the startDir directory and moving on to the
+  // parent directory recursively (going no further than the root directory)
+  // find and return the path to the first encountered tsconfig.json file
+
+  if (!startDir) {
+    return getPathToClosestTSConfig(getStartDir());
+  }
+
+  const tsConfigPath = path.join(startDir, 'tsconfig.json');
+
+  const startDirPath = path.resolve(startDir);
+  const previousDirPath = path.resolve(previousDir || '/');
+
+  if (startDirPath === previousDirPath || fs.existsSync(tsConfigPath)) {
```

Comment:
So if I understand this right - we're travelling all the way to the root of the file system looking for .tsconfigs? It's a little hard to tell from the naming when this algorithm stops.

Either way, currently it seems like if we can't resolve the tsconfig, we just return a faulty (root) path - I woud much rather throw an error

Reply:
It will throw an error when it tries to open a non existent tsconfig.json.

Reply:
Fair enough.